apps/web/src/components/Dashboard/Templates.tsx

```diff
@@ -28,43 +59,128 @@ export function TemplatesContent({
+          },
+        }
+      )
+
+      const data: Template[] = await res.json()
+      if (!res.ok) {
+        toast({
+          title: 'An error occurred',
+          description: 'We were unable to fetch the team members',
         })
+        console.log(res.statusText)
+        return
       }
-    }
 
-    const interval = setInterval(() => {
-      f()
-    }, 5000)
+      // Display the most recently updated templates first
+      const orderedData = data.sort((a, b) => {
+        return new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime()
+      })
+      setTemplates(orderedData)
+    }
 
-    f()
-    // Cleanup interval on component unmount
-    return () => clearInterval(interval)
+    fetchTemplates()
   }, [user, teamId])
 
+  async function deleteTemplate() {
+    const res = await fetch(
+      `${process.env.NEXT_PUBLIC_API_URL}/templates/${currentTemplate?.templateID}`,
+      {
+        method: 'DELETE',
+        headers: {
+          Authorization: `Bearer ${user.accessToken}`,
+        },
+      }
+    )
+
+    if (!res.ok) {
+      return toast({
+        title: 'An error occurred',
+        description: 'We were unable to delete the template',
+      })
+    }
+
+    setTemplates(
+      templates.filter(
+        (template) => template.templateID !== currentTemplate?.templateID
+      )
+    )
+    setCurrentTemplate(null)
+    setIsDeleteTemplateDialogOpen(false)
+  }
+
+  async function publishTemplate() {
+    const res = await fetch(
+      `${process.env.NEXT_PUBLIC_API_URL}/templates/${currentTemplate?.templateID}`,
+      {
+        method: 'PATCH',
+        headers: {
+          Authorization: `Bearer ${user.accessToken}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          public: !currentTemplate?.public,
+        }),
+      }
+    )
+
+    if (!res.ok) {
+      return toast({
+        title: 'An error occurred',
+        description: 'We were unable to update the template',
+      })
+    }
+
+    toast({
+      title: 'Template updated',
+      description: 'The template has been updated successfully',
+    })
+
+    setTemplates(
+      templates.map((template) =>
+        template.templateID === currentTemplate?.templateID
+          ? { ...template, public: !template.public }
+          : template
+      )
+    )
+    setCurrentTemplate(null)
+    setIsPublishTemplateDialogOpen(false)
+  }
+
   return (
     <div className="flex flex-col justify-center">
       <Table>
         <TableHeader>
-          <TableRow className="hover:bg-orange-500/10 dark:hover:bg-orange-500/10 border-b border-white/5 ">
+          <TableRow className="hover:bg-orange-500/10 dark:hover:bg-orange-500/10 border-b border-white/5">
+            <TableHead>Visibility</TableHead>
             <TableHead>Template ID</TableHead>
-            <TableHead>Template Name</TableHead>
+            <TableHead>Aliases</TableHead>
```

Comment:
is `Alias` consistent with the rest of SDK's naming convention?

Reply:
"Template Name" is the one we're using currently in the CLI, but in SDKs it's alias

Reply:
I see `Template Name` in CLI and in SDK it isn't really exposed

Reply:
Ok, reverted